controller/crds/amaltheasession.yaml

```diff
@@ -0,0 +1,272 @@
+                              type: string
+                          required:
+                          - name
+                          - value
+                      image:
+                        type: string
+                      memoryLimit:
+                        type: string
+                      cpuLimit:
+                        type: string
+                      name:
+                        type: string
+                    required:
+                    - name
+                    - image
+
+                initContainers:
+                  type: array
+                  default: []
+                  description: Sidecar containers to run along the main cotnainer
+                  items:
+                    type: object
+                    properties:
+                      args:
+                        type: array
+                        description: Arguments passed to the command of the image or the
+                          overridden command below.
+                        items:
+                          type: string
+                      command:
+                        type: array
+                        description: Command to run in place of the image provided
+                          command.
+                        items:
+                          type: string
+                      env:
+                        type: array
+                        description: Environment variables used in this container
+                        items:
+                          type: object
+                          properties:
+                            name:
+                              type: string
+                            value:
+                              type: string
+                          required:
+                          - name
+                          - value
+                      image:
+                        type: string
+                      memoryLimit:
+                        type: string
+                      cpuLimit:
+                        type: string
+                      name:
+                        type: string
+                    required:
+                    - name
+                    - image
+
+                dataSources:
+                  type: array
+                  default: []
+                  description: List of data source to connect to and mount in the main
+                    container.
+                  items:
+                    type: object
+                    properties:
+                      remote:
+                        type: string
+                        description: URL of the data source to connect to
+                      mountPath:
+                        type: string
+                        description: Path to were the data source should be mounted
+                      secret:
+                        type: string
+                        description: Name of the secret containing the credentials for the
```

Comment:
I am not sure how else we can generalize this. If we have no constraints on this secret at all it becomes unusable.

Reply:
The alternative is to ask for a list of pvc names and paths where they should be mounted in the session. Then we do not have to ask for a secret and it is the job of the user of amalthea to create the PVCs whichever way they see fit.

Reply:
Since a data source might be different things, what about having an rclone object a bit like the git object from the repositories above ?

That way, it makes it easier to add support for other data sources.

Reply:
@sgaist this makes sense. I will add this when I convert the CRD spec to the go framework we will use. You dont have to add it now just so we change it later.